kubetest2-gke/deployer/deployer.go

```diff
@@ -223,56 +240,84 @@ func (d *deployer) Up() error {
 	if err := d.init(); err != nil {
 		return err
 	}
-	if err := d.prepareGcpIfNeeded(); err != nil {
+
+	// Only run prepare once for the first GCP project.
+	if err := d.prepareGcpIfNeeded(d.projects[0]); err != nil {
 		return err
 	}
-
-	// Create network if it doesn't exist.
-	if runWithNoOutput(exec.Command("gcloud", "compute", "networks", "describe", d.network,
-		"--project="+d.project,
-		"--format=value(name)")) != nil {
-		// Assume error implies non-existent.
-		log.Printf("Couldn't describe network '%s', assuming it doesn't exist and creating it", d.network)
-		if err := runWithOutput(exec.Command("gcloud", "compute", "networks", "create", d.network,
-			"--project="+d.project,
-			"--subnet-mode=auto")); err != nil {
-			return err
-		}
+	if err := d.createNetwork(); err != nil {
+		return err
+	}
+	if err := d.setupNetwork(); err != nil {
+		return err
 	}
 
+	log.Printf("Environment: %v", os.Environ())
+	eg, _ := errgroup.WithContext(context.Background())
```

Comment:
nice! but we will probably need to get the Contextwithcancel here to propagate to all the cluster create calls.

Reply:
Done. I have changed to use `context.WithCancel` and call the `cancel` method whenever one cluster creation process fails.